ldes-server-compaction/src/main/java/be/vlaanderen/informatievlaanderen/ldes/server/compaction/application/services/FragmentCompactionService.java

```diff
@@ -0,0 +1,44 @@
+package be.vlaanderen.informatievlaanderen.ldes.server.compaction.application.services;
+
+import be.vlaanderen.informatievlaanderen.ldes.server.domain.model.FragmentPair;
+import be.vlaanderen.informatievlaanderen.ldes.server.domain.model.LdesFragmentIdentifier;
+import be.vlaanderen.informatievlaanderen.ldes.server.fragmentation.entities.Fragment;
+import be.vlaanderen.informatievlaanderen.ldes.server.fragmentation.repository.FragmentRepository;
+import org.springframework.stereotype.Service;
+
+import java.util.ArrayList;
+import java.util.List;
+
+@Service
+public class FragmentCompactionService {
+	public static final String PAGE_NUMBER = "pageNumber";
+	private final FragmentRepository fragmentRepository;
+	private final CompactedFragmentCreator compactedFragmentCreator;
+
+	public FragmentCompactionService(FragmentRepository fragmentRepository,
+			CompactedFragmentCreator compactedFragmentCreator) {
+		this.fragmentRepository = fragmentRepository;
+		this.compactedFragmentCreator = compactedFragmentCreator;
+	}
+
+	public void compactFragments(Fragment firstFragment, Fragment secondFragment) {
+		LdesFragmentIdentifier ldesFragmentIdentifier = generateNewLdesFragmentIdentifier(firstFragment,
+				secondFragment);
+		if (fragmentRepository.retrieveFragment(ldesFragmentIdentifier).isEmpty()) {
```

Comment:
Maybe we can add warning logging when this is false? Just in case we have a bug somewhere and end up in an infinite loop? Especially when we move to UUID pageNumbers

Reply:
Done